internal/graph/graph.go

```diff
@@ -332,6 +335,61 @@ nextSample:
 	return selectNodesForGraph(nodes, o.DropNegative)
 }
 
+// Trims a Graph that is in forest form to contain only the nodes in kept. This
+// will not work correctly in the case that a node has multiple parents.
+func (g *Graph) TrimTree(kept NodeSet) *Graph {
```

Comment:
There is no need to return a *Graph, and in fact it makes the interface confusing:
- Does it return a new graph? Does it update the incoming graph?